crates/variables/src/lib.rs

```diff
@@ -163,31 +163,27 @@ impl<'a> AsRef<str> for Key<'a> {
 
 type Result<T> = std::result::Result<T, Error>;
 
-/// A config resolution error.
+/// A variable resolution error.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    /// Invalid config key.
-    #[error("invalid config key: {0}")]
+    /// Invalid variable key.
+    #[error("invalid variable key: {0}")]
```

Comment:
key -> name now?

Reply:
This tempts me to rename the `Key` wrapper struct to `VariableName`. Thoughts or am I just OCD?

Reply:
I'd probably leave it for now.  It's purely internal, it's not doing any harm, and it would create even more churn in a PR that already has wide impact.  But conversely, it's a fairly localised change so won't make rebases around this PR any harder.  So if you are keen then go for it!